docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,411 @@
+# test and trial functions (among other things). To define
+# these we need to specify an interpolation space for the shape functions.
+# We use `DiscontinuousLagrange` functions
+# based on the two-dimensional reference quadrilateral. We also define a quadrature rule based on
+# the same reference element. We combine the interpolation and the quadrature rule
+# to `CellValues` and `InterfaceValues` object. Note that `InterfaceValues` object contains two `FaceValues` objects which can be used individually.
+ip = DiscontinuousLagrange{RefQuadrilateral, 1}()
+qr = QuadratureRule{RefQuadrilateral}(2)
+# For `FaceValues` and `InterfaceValues` we use `FaceQuadratureRule`
+face_qr = FaceQuadratureRule{RefQuadrilateral}(2)
+cellvalues = CellValues(qr, ip);
+facevalues = FaceValues(face_qr, ip);
+interfacevalues = InterfaceValues(facevalues)
+# ### Degrees of freedom
+# Next we need to define a `DofHandler`, which will take care of numbering
+# and distribution of degrees of freedom for our approximated fields.
+# We create the `DofHandler` and then add a single scalar field called `:u` based on
+# our interpolation `ip` defined above.
+# Lastly we `close!` the `DofHandler`, it is now that the dofs are distributed
+# for all the elements.
+dh = DofHandler(grid)
+add!(dh, :u, ip)
+close!(dh);
+
+# Now that we have distributed all our dofs we can create our tangent matrix,
+# using `create_sparsity_pattern`. This function returns a sparse matrix
+# with the correct entries stored. We need to pass the topology and the cross-element coupling matrix when we're using
+# discontinuous interpolations. The cross-element coupling matrix is of size [1,1] in this case as
+# we have only one field and one DofHandler.
+K = create_sparsity_pattern(dh, topology = topology, cross_coupling = trues(1,1))
+
+# ### Boundary conditions
+# In Ferrite constraints like Dirichlet boundary conditions
+# are handled by a `ConstraintHandler`.
+ch = ConstraintHandler(dh);
+
+# Next we need to add constraints to `ch`. For this problem we define
+# inhomogeneous Dirichlet boundary conditions on the two boundaries.
+# We define $\partial \Omega_n$ as the `union` of the face sets with Neumann boundary conditions.
+∂Ωₙ = union(
+    getfaceset(grid, "top"),
+    getfaceset(grid, "bottom"),
+);
+
+# Now we are set up to define our constraint. We specify which field
+# the condition is for, and our face sets. The last
+# argument is a function of the form $f(\textbf{x})$ or $f(\textbf{x}, t)$,
+# where $\textbf{x}$ is the spatial coordinate and
+# $t$ the current time, and returns the prescribed value.
+add!(ch, Dirichlet(:u, getfaceset(grid, "right"), (x, t) -> 1.0));
+add!(ch, Dirichlet(:u, getfaceset(grid, "left"), (x, t) -> -1.0));
+
+# Finally we also need to `close!` our constraint handler. When we call `close!`
+# the dofs corresponding to our constraints are calculated and stored
+# in our `ch` object.
+close!(ch)
```

Comment:
Since this is an advanced example, perhaps it suffices to have something like 
```suggestion
# ### Boundary conditions
# The Dirichlet boundary conditions are treated 
# as usual by a `ConstraintHandler`.
ch = ConstraintHandler(dh)
add!(ch, Dirichlet(:u, getfaceset(grid, "right"), (x, t) -> 1.0))
add!(ch, Dirichlet(:u, getfaceset(grid, "left"), (x, t) -> -1.0))
close!(ch);

# Furthermore, we define $\partial \Omega_n$ as the `union` of the face sets with Neumann boundary conditions for later use
∂Ωₙ = union(
    getfaceset(grid, "top"),
    getfaceset(grid, "bottom"),
);
```